### DISCUSSION TITLE
Size Limits of Automatic Export

### DISCUSSION THREAD

**Discussion by stefanopagliari** [retorquere/zotero-better-bibtex#1842] [Q&A]
My workflow to collect/annotate/process literature notes currently consists of a mix of different plugins:
- Better bittex: to generate the unique citekey
- Zotfile: to export the annotations and give the pdf the same name as the bbt citekey
- mdnote: to export the metadata and annotations into a md file

I would like to take advantage of the automatic export function of BBT to automate the process of creating new literature notes once a reference is added or edited, in conjunction with a script that would break the bib file in different md files, one for each reference. 

While the automatic export of only the metadata of my library (3000 items) is very fast, I'm wondering what will happen once the annotations and highlights are added and the amount of text increase (I currently have limited annotations exported from PDF so I can't test it).
Would the automatic export would become to long once the size of the bib file grows from 1.4mb (current size) to tens or hundreads of MB? 
Is there any way to tweak the setting so that each reference is exported into a separate bib file?

I realize this is not the use case that BBT is built for, and in many ways what I need is the automatic export function added to MDNotes.

**Comment by retorquere** [comment]
That's not simply a settings tweak, no.

One of my tests exports 24k items and that works alright.

I'm also working on a project that will allow using the BBT exporters against yout online library without Zotero running. That should be able to do what you want fairly smoothly.

**Reply by stefanopagliari** [reply]
In my case the number of items is unlikely to grow that much, but the size of the text to be exported will once the highlights and annotations are added (e.g. consider all the highlighted sentence from a book). 

That project sounds very interesting. I will keep an eye on that

**Reply by stefanopagliari** [reply]
I have one quick follow-up question. Am I right in understanding that when the automatic export is "on change", this is triggered by changes of the metadata but not by changes in the tags or collections of a reference?

**Reply by retorquere** [reply]
Both changes in the item data and tags and I think also attachments being added/removed. Not which collections an item belongs to. I simply get notified "item has changed in some way" and I don't get details on what changed about it, but I get separate notifications for "item added to/removed from collection".
